WcaOnRails/app/webpacker/components/CompetitionOverview/useDelegatesData.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+import { delegatesApiUrl, WCA_API_PAGINATION } from '../../lib/requests/routes.js.erb';
+import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
+
+const useDelegatesData = () => {
+  const [delegatesData, setDelegatesData] = useState([]);
+
+  const {
+    data: rawDelegatesData,
+    fetchNextPage: delegateFetchNextPage,
+    hasNextPage: delegateHasNextPage,
```

Comment:
Since this is isolated inside the `useDelegatesData` hook, it's kind of implicit that everything is a delegate, and there are no other queries to potential mix up, so you could just use the original names `data`, `fetchNextPage`, and `hasNextPage` to (slightly) shorten some code.